code/controllers/subsystem/mapping.dm

```diff
@@ -552,8 +552,9 @@ GLOBAL_LIST_EMPTY(the_station_areas)
 				return reserve
 		//If we didn't return at this point, theres a good chance we ran out of room on the exisiting reserved z levels, so lets try a new one
 		num_of_res_levels += 1
-		var/newReserved = add_new_zlevel("Transit/Reserved [num_of_res_levels]", list(ZTRAIT_RESERVED = TRUE))
-		if(reserve.Reserve(width, height, newReserved))
+		var/datum/space_level/newReserved = add_new_zlevel("Transit/Reserved [num_of_res_levels]", list(ZTRAIT_RESERVED = TRUE))
+		initialize_reserved_level(newReserved.z_value)
+		if(reserve.Reserve(width, height, newReserved.z_value))
```

Comment:
Now passes ``newReserved.z_value`` instead of just ``newReserved``

Reply:
Proc prototype is ``/datum/turf_reservation/proc/Reserve(width, height, zlevel)``, so is probably correct

Reply:
z_value is defined on newReserved